lib/Runtime/Types/Type.h

```diff
@@ -62,6 +63,14 @@ namespace Js
         inline BOOL SkipsPrototype() const { return (this->flags & TypeFlagMask_SkipsPrototype) != 0 ; }
         inline BOOL CanHaveInterceptors() const { return (this->flags & TypeFlagMask_CanHaveInterceptors) != 0; }
         inline BOOL IsFalsy() const { return flags & TypeFlagMask_IsFalsy; }
+        inline BOOL HasBeenCached() const { return flags & TypeFlagMask_HasBeenCached; }
+        inline void SetHasBeenCached(const bool hasBeenCached)
+        {
+            // Once set, this flag should never be reset.
```

Comment:
Not changing the type on a delete property if the type wasn't cached is an optimization, not a functional change. So, in the case that you brought up, we'll just behave the same as we do now.

---
In reply to: [78818005](https://github.com/Microsoft/ChakraCore/pull/1542#discussion_r78818005) [](ancestors = 78818005,78816821,78790053)